### PR TITLE
Link to home page in TOC and update extensions

### PR DESCRIPTION
**Pull request by CalMacCQ**
I've added the API docs home to the docs sidebar following -> https://stackoverflow.com/questions/16123951/how-do-i-include-the-homepage-in-the-sphinx-toc

Screenshot 2023-11-22 at 13 47 57

I've also added the qulacs density matrix simulation to the extensions index.

### CODE REVIEW

File: pytket/docs/index.rst [hunk 1]
```diff
@@ -75,18 +75,19 @@ Licensed under the `Apache 2 License <http://www.apache.org/licenses/LICENSE-2.0
...
 .. toctree::
-    :caption: Introduction:
+    :caption: Overview:
     :maxdepth: 1
 
+    API docs home <self> 
     getting_started.rst
     changelog.rst
     install.rst
+    TKET website <https://tket.quantinuum.com/>
```
Comment (cqc-melf) [inline pytket/docs/index.rst:85]: Why are we not adding extension, examples and manual to this overview?

Comment (CalMacCQ) [reply pytket/docs/index.rst:85]: Do you think its more apporiate to merge the "Overview" and "More documentation" lists?

Comment (cqc-melf) [reply pytket/docs/index.rst:85]: Yes, but if you have good reasons to not do that, I am happy to discuss that. But I am not sure what our general plan is how and if we want to add the general navigation bar on top?

Comment (CalMacCQ) [reply pytket/docs/index.rst:85]: I was thinking we add these links here as an easy near term solution and then figure out the navbar later on.

**Pull request closed without merging**